autogpt_platform/frontend/src/hooks/useAgentGraph.ts

```diff
@@ -405,7 +405,34 @@ export default function useAgentGraph(
             setNestedProperty(errors, key, "This field is required");
           }
         });
-      }
+      } 
+      
+      Object.entries(node.data.inputSchema.properties || {}).forEach(([key, schema]) => {
```

Comment:
Since you're doing this validation on save, it should be done in the backend. Frontend-only validation = vulnerable.